test/cog/command/service/data_store_test.exs

```diff
@@ -0,0 +1,68 @@
+defmodule Cog.Command.Service.DataStoreTest do
+  use ExUnit.Case, async: true
+
+  alias Cog.Command.Service.DataStore
+
+  @base_path Application.get_env(:cog, Cog.Command.Service)[:data_path]
+  @namespace [ "test", "data_store" ]
+
+  setup_all do
+    Ecto.Adapters.SQL.Sandbox.checkout(Cog.Repo)
+    Ecto.Adapters.SQL.Sandbox.mode(Cog.Repo, {:shared, self()})
+
+    pid = Process.whereis(DataStore)
+    Process.unregister(DataStore)
+
+    on_exit(fn ->
+      Process.register(pid, DataStore)
+    end)
+
+    :ok
+  end
+
+  setup do
+    {:ok, pid} = DataStore.start_link(@base_path)
+    {:ok, %{pid: pid}}
+  end
+
+  test "fetch an existing key" do
+    foo = %{"foo" => "fooval"}
+    DataStore.replace(@namespace, "foo", foo)
+    assert {:ok, ^foo} = DataStore.fetch(@namespace, "foo")
+  end
+
+  test "fetch an unset key" do
+    assert {:error, "Object not found"} = DataStore.fetch(@namespace, "cheeseburgers")
+  end
+
+  test "replacing an unset key" do
+    foo1 = %{"foo1" => "fooval"}
+    DataStore.replace(@namespace, "foo1", foo1)
+    assert {:ok, ^foo1} = DataStore.fetch(@namespace, "foo1")
+    DataStore.delete(@namespace, "foo1")
```

Comment:
It's mostly there to keep cruft from accumulating after tests since files get written to the filesystem and we don't have transactions to roll them back. I can use an `on_exit` instead.